spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
```

Comment:
Our use of a reentrant mutex is an implementation detail, IMO, and isn't part of the public facing behavior of RSpec.  I'd phrase this as:

``` ruby
it 'avoids deadlocks when a proxy is accessed from within a `before_verifying_doubles` callback'
```